modules/javafx.controls/src/main/java/javafx/scene/control/skin/TextAreaSkin.java

```diff
@@ -847,76 +798,43 @@ public Bounds getCaretBounds() {
-        int n = paragraphNodes.getChildren().size();
-
-        int paragraphIndex = 0;
-        int offset = index;
-
-        String paragraph = null;
-        while (paragraphIndex < n) {
-            Text paragraphNode = (Text)paragraphNodes.getChildren().get(paragraphIndex);
-            paragraph = paragraphNode.getText();
-            int count = paragraph.length() + 1;
-
-            if (offset < count) {
-                break;
-            }
-
-            offset -= count;
-            paragraphIndex++;
-        }
-
-        return offset == paragraph.length() ? '\n' : paragraph.charAt(offset);
+        String paragraph = getTextNode().getText();
+        return index == paragraph.length() ? '\n' : paragraph.charAt(index);
     }
 
     /** {@inheritDoc} */
-    @Override protected int getInsertionPoint(double x, double y) {
+    @Override
+    protected int getInsertionPoint(double x, double y) {
         TextArea textArea = getSkinnable();
-
-        int n = paragraphNodes.getChildren().size();
-        int index = -1;
-
-        if (n > 0) {
-            if (y < contentView.snappedTopInset()) {
-                // Select the character at x in the first row
-                Text paragraphNode = (Text)paragraphNodes.getChildren().get(0);
-                index = getNextInsertionPoint(paragraphNode, x, -1, VerticalDirection.DOWN);
-            } else if (y > contentView.snappedTopInset() + contentView.getHeight()) {
-                // Select the character at x in the last row
-                int lastParagraphIndex = n - 1;
-                Text lastParagraphView = (Text)paragraphNodes.getChildren().get(lastParagraphIndex);
-
-                index = getNextInsertionPoint(lastParagraphView, x, -1, VerticalDirection.UP)
-                        + (textArea.getLength() - lastParagraphView.getText().length());
-            } else {
-                // Select the character at x in the row at y
-                int paragraphOffset = 0;
-                for (int i = 0; i < n; i++) {
-                    Text paragraphNode = (Text)paragraphNodes.getChildren().get(i);
-
-                    Bounds bounds = paragraphNode.getBoundsInLocal();
-                    double paragraphViewY = paragraphNode.getLayoutY() + bounds.getMinY();
-                    if (y >= paragraphViewY
-                            && y < paragraphViewY + paragraphNode.getBoundsInLocal().getHeight()) {
-                        index = getInsertionPoint(paragraphNode,
-                                x - paragraphNode.getLayoutX(),
-                                y - paragraphNode.getLayoutY()) + paragraphOffset;
-                        break;
-                    }
-
-                    paragraphOffset += paragraphNode.getText().length() + 1;
-                }
+        Text paragraphNode = getTextNode();
+
+        if (y < contentView.snappedTopInset()) {
+            // Select the character at x in the first row
+            return 0;
```

Comment:
I see that the implementation of the now-removed `getNextInsertionPoint` method also returned `0` unconditionally, so this is equivalent, but... isn't the comment wrong, since `x` isn't used in this case?

Reply:
comment is indeed wrong, will fix.